pkg/crypto/ed25519.go

```diff
@@ -0,0 +1,111 @@
+package crypto
+
+import (
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"os"
+
+	"golang.org/x/crypto/ed25519"
+)
+
+var (
+	// PubKeyIdentifier is the PEM public key identifier
+	PubKeyIdentifier = "PUBLIC KEY"
+	// PrivKeyIdentifier is the PEM private key identifier
+	PrivKeyIdentifier = "PRIVATE KEY"
+	// PEMCipher is the PEM encryption algorithm
+	PEMCipher = x509.PEMCipherAES256
+	// PubKeyFilePermissions are the public key file perms
+	PubKeyFilePermissions os.FileMode = 0644
+	// PrivKeyFilePermissions are the private key file perms
+	PrivKeyFilePermissions os.FileMode = 0600
+)
+
+// LoadPublicKeyFromFile loads PEM formatted ED25519 public key from file.
+func LoadPublicKeyFromFile(publicKeyPath string) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(publicKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	var block *pem.Block
+	for {
+		block, _ = pem.Decode(x509PEM)
+		if block != nil && block.Type == PubKeyIdentifier {
+			break
+		} else if block == nil {
+			return nil, errors.New("Can't decode PEM file")
+		}
+	}
+
+	return block.Bytes, nil
+}
+
+// LoadPrivateKeyFromFile loads PEM formatted ED25519 private key from file.
+func LoadPrivateKeyFromFile(privateKeyPath string, password []byte) ([]byte, error) {
+	x509PEM, err := ioutil.ReadFile(privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// Parse x509 PEM file
+	var block *pem.Block
+	for {
+		block, _ = pem.Decode(x509PEM)
```

Comment:
same as above. This should be actually moved into a separate function and called by LoadPublicKeyFromFile and LoadPrivateKeyFromFile